libs/protocols/src/threshold_eddsa/state.rs

```diff
@@ -0,0 +1,237 @@
+
+use rand::rngs::OsRng;
+use state_machine::{
+    state::{Recipient, RecipientMessage, StateMachineMessage},
+    StateMachineStateExt, StateMachineStateOutput, StateMachineStateResult,
+};
+use state_machine_derive::StateMachineState;
+
+use shamir_sharing::party::PartyId;
+
+/// Threshold EdDSA round1 state machine.
+
+pub mod states {
+    use basic_types::jar::PartyJar;
+    use cggmp21::generic_ec::curves::Ed25519;
+    use givre::signing::{
+        round1::{PublicCommitments, SecretNonces},
+        round2::SigShare,
+    };
+    use key_share::{CoreKeyShare, KeyInfo};
+
+    use crate::threshold_ecdsa::util::SortedParties;
+    /// The protocol is waiting for the pre processing phase to finish
+    /// and to receive a message to sign
+    pub struct WaitingPublicCommits {
+        /// Key Share
+        pub(crate) key_share: CoreKeyShare<Ed25519>,
+        /// Nonces from pre processing phase
+        pub(crate) secret_nonces: SecretNonces<Ed25519>,
+        /// Vector of Public Commitments from pre processing phase
+        pub(crate) pcommits: PartyJar<PublicCommitments<Ed25519>>,
+        /// Message to sign
+        pub(crate) msg: Vec<u8>,
+        /// Sorted Parties
+        pub(crate) sorted_parties: SortedParties,
+    }
+    /// The protocol is waiting for the signatures shares
+    /// from each signer
+    pub struct WaitingSigShares {
+        /// Public Key Info
+        pub(crate) key_info: KeyInfo<Ed25519>,
+        /// Vector of Public Commitments from pre processing phase
+        pub(crate) pcommits: PartyJar<PublicCommitments<Ed25519>>,
+        /// Vector with the Signatures Shares from each signer
+        pub(crate) sig_shares: PartyJar<SigShare<Ed25519>>,
+        /// Message to sign
+        pub(crate) msg: Vec<u8>,
+        /// Sorted Parties
+        pub(crate) sorted_parties: SortedParties,
+    }
+}
+
+/// The state machine for the EdDSA protocol.
+#[derive(StateMachineState)]
+#[state_machine(
+    recipient_id = "PartyId",
+    input_message = "PartyMessage<Msg<Ed25519>>",
+    output_message = "Msg<Ed25519>",
+    final_result = "EddsaSignatureOutput",
+    handle_message_fn = "Self::handle_message"
+)]
+pub enum EddsaSignState {
+    /// We are waiting for the Public Commits
+    #[state_machine(completed = "state.pcommits.is_full()", transition_fn = "Self::transition_waiting_public_commits")]
+    WaitingPublicCommits(states::WaitingPublicCommits),
+    /// We are waiting for the SigShares
+    #[state_machine(completed = "state.sig_shares.is_full()", transition_fn = "Self::transition_waiting_sigshares")]
+    WaitingSigShares(states::WaitingSigShares),
+}
+
+use EddsaSignState::*;
+/// Edd
+impl EddsaSignState {
+    /// af
+    pub fn new(
+        parties: Vec<PartyId>,
+        msg: Vec<u8>,
+        incomplete_key_share: ThresholdPrivateKeyShare<Ed25519>,
+    ) -> Result<(Self, Vec<StateMachineMessage<Self>>), EddsaSignError> {
+        // Compute input elements required for their state machine
```

Comment:
```suggestion
        // Compute input elements required for givre's functions
```
We are not using their state machines